source/blog/2017-12-02-installing-etcd-as-a-system-container.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Installing etcd as a System container
+author: smilner
+date: 2017-10-20 20:00:00 UTC
+published: false
+comments: true
+tags: atomic, etcd, system containers, runc, containers
+---
+
+System Containers exist as a way to provide containerized services to a host before traditional container runtimes are ready. System
+Containers do this by utilizing a runc and systemd units and do not deviate from the OCI standards. Let's look at how someone can
+use a system containers to install the ever popular etcd today one Fedora 27 Atomic Host!
+
+READMORE
+
+
+## Atomic Host
+
+The first thing one must have is Fedora 27 Atomic Host installed. As a reminder, Atomic Host allows people to:
+
+> Use immutable infrastructure to deploy and scale your containerized applications. Project Atomic builds OSes, tools, and containers for cloud native platforms.
+
+To grab a copy of the Fedora 27 version of Atomic Host head on over to the [get fedora page for atomic](https://getfedora.org/en/atomic/download/).
+
+
+## Installing etcd
+
+As one would expect the ``atomic`` command is used for pulling, installing, updating, deleting, and manipulating system containers.
+Let's install the [etcd system container](https://admin.fedoraproject.org/pkgdb/package/container/etcd/) from the
+[Fedora Layered Image Build System](https://docs.pagure.org/releng/layered_image_build_service.html) with the following command:
+
+```
+$ sudo atomic install --system --storage ostree --name=etcd registry.fedoraproject.org/f26/etcd:latest
+Pulling layer 01aae00bf9e4a7301133bac6641015fc1677a19ef13844f5b274cf6233515fdf
+Pulling layer 9dbaf15c249f7ed8a76b8f2b785cc603172af8271e18cf28884bca36f7e39311
+Extracting to /var/lib/containers/atomic/etcd.0
+Created file /etc/etcd/etcd.conf
+Created file /usr/local/bin/etcdctl
+systemctl daemon-reload
+systemd-tmpfiles --create /etc/tmpfiles.d/etcd.conf
+systemctl enable etcd
+$
+```
+
+What did we do? Let's break it down:
+
+* ``sudo atomic``: We are executing the atomic command with root privledges
+* ``install``: We are denoting we want to install a container
+* ``--system``: And the type of the container we want to install is a system container
+* ``--storage ostree``: We'd like the image to be stored in ostree
+* ``--name etcd``: The name of the container should be etcd
+* ``registry.fedoraproject.org/f26/etcd:latest``: This is the image to use when creating the container
```

Comment:
could you add some notes about `--set` and how some images, etcd in this case, allow a configuration this way?

It is also possible to see what variables can be customized: `atomic info --storage ostree $IMG` shows all the supported variables.
With the etcd system container these settings can be overriden anyway by the etcd.conf file, that we now bind mount from the host.

Reply:
Done